xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/annotation/PropertyDisplayType.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.properties.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Use this annotation to specify the class type of a property.
+ *
+ * @version $Id$
+ * @since 11.0RC1
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.FIELD })
+@Inherited
+public @interface PropertyDisplayType
```

Comment:
This is not enough IMO since you only cover Class use case. Problem is that Type is not allowed in an annotation but at the very least you should add optional Class[] parameters field to cover things like List<Integer> (and update descriptor code accordingly), that should be enough for 99% use cases.

Reply:
I've made the improvement, thanks for the suggestion.

Reply:
I feel having everything in `values` might be a bit confusing. At the very least the javadoc should be a bit more details on how to use it depending on the need.

Reply:
I agree that the javadoc should be improved, at least with examples, but I think this is actually better than adding another field. For instance you can write:
```
@PropertyDisplayType(Boolean.class)
```
=> `Boolean`
and
```
@PropertyDisplayType({List.class, Boolean.class})
```
=> `List<Boolean>`